### PR TITLE
feat: CRN-1196 Update Algolia CLI

### PR DESCRIPTION
**Pull request by peterstarling**
## Notable changes
* no longer store Algolia backup, copy dev index for new PRs instead
* change entity export script to parse to Algolia format directly
* remove deprecated Algolia cli

### CODE REVIEW

File: .github/workflows/on-schedule-backup-dev.yml [hunk 1]
```diff
@@ -55,46 +55,3 @@ jobs:
...
-      - name: Get Algolia Keys
-        id: algolia-keys
-        uses: ./.github/actions/algolia-keys
-        with:
-          algolia-app-id-name: ${{ steps.setup.outputs.algolia-app-id-name }}
-          algolia-api-key-name: ${{ steps.setup.outputs.algolia-api-key-name }}
-          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
-          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
-          aws-default-region: ${{ steps.setup.outputs.aws-default-region }}
-      - name: Run the backup
```
Comment (kmaid) [inline .github/workflows/on-schedule-backup-dev.yml:84]: So now we don't backup Algolia. We just reindex?

File: .github/workflows/reusable-algolia-sync.yml [hunk 2]
```diff
@@ -39,6 +39,8 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@v3
+      - name: Install Algolia CLI
+        uses: algolia/setup-algolia-cli@master
```
Comment (kmaid) [inline .github/workflows/reusable-algolia-sync.yml:43]: Nice

File: apps/crn-server/scripts/cli.ts [hunk 3]
```diff
@@ -13,7 +13,7 @@ yargs(hideBin(process.argv))
         .positional('entity', {
           describe: 'specific an entity to import',
           type: 'string',
-          choices: ['users', 'research-outputs', 'external-authors', 'events'],
+          choices: ['user', 'research-output', 'external-author', 'event'],
```
Comment (kmaid) [inline apps/crn-server/scripts/cli.ts:16]:
```suggestion
          choices: <const>['user', 'research-output', 'external-author', 'event'],
```

Comment (kmaid) [inline apps/crn-server/scripts/cli.ts:16]: failing that 
```suggestion
          choices: ['user', 'research-output', 'external-author', 'event'] as const[], 
```

File: apps/crn-server/scripts/export-entity.ts [hunk 4]
```diff
@@ -114,15 +121,27 @@ function getController(entity: Entity) {
...
+    'external-author': new ExternalAuthors(squidexGraphqlClient),
+    event: new Events(squidexGraphqlClient, eventsRestClient),
+    lab: new Labs(squidexGraphqlClient),
   };
 
   return controllerMap[entity];
-}
+};
+
+const transformRecords = <T extends keyof EntityResponses>(
```
Comment (kmaid) [inline apps/crn-server/scripts/export-entity.ts:138]: So we are still doing the transform step rather than adding it to the entity?

**Review by kmaid — APPROVED** [review]

File: .github/workflows/on-schedule-backup-dev.yml (hunk 1 above)
Comment (peterstarling) [reply .github/workflows/on-schedule-backup-dev.yml:84]: yeh

File: apps/crn-server/scripts/export-entity.ts (hunk 4 above)
Comment (peterstarling) [reply apps/crn-server/scripts/export-entity.ts:138]: yeh at least here it was easier

**Review by markcsims — APPROVED** [review]

**Pull request merged**